tests/js/server/aql/aql-optimizer-geoindex.js

```diff
@@ -1029,7 +1029,64 @@ function optimizerRuleTestSuite() {
       var result = AQL_EXPLAIN(query.string, query.bindVars);
       hasIndexNode(result, query);
       hasNoFilterNode(result, query);
-    }
+    },
+
+    // Regression test (https://arangodb.atlassian.net/browse/AR-113):
+    // Previously, the filter was moved into the index, but one access of `dist` in `dist < dist` was left, while the
+    // variable obviously can't be available yet at the index node.
+    // When failing, throws an error like
+    //   missing variable #1 (dist) for node #8 (IndexNode) while planning registers
+    testSelfReference1: function () {
+      const query = {
+        string: `
+          FOR doc IN @@col
+            LET dist = GEO_DISTANCE(doc.geo, [16, 53])
+            FILTER dist < dist
```

Comment:
I would think this would simply get optimized to `FILTER false` and isn't a good test.

Reply:
This is the query @Simran-B reported in [AR-113](https://arangodb.atlassian.net/browse/AR-113), and, without the bug fix, actually results in
> Query: AQL: missing variable #1 (dist) for node #8 (IndexNode) while planning registers (exception location: /work/ArangoDB/arangod/Aql/RegisterPlan.cpp:174). Please report this error to arangodb.com (while optimizing plan) (exception location: /work/ArangoDB/arangod/RestHandler/RestCursorHandler.cpp:305). Please report this error to arangodb.com
.

Reply:
Fair enough. I'd consider this an excellent opportunity to add an optimization to simplify some expressions then! (In another PR of course.)

Reply:
FILTER conditions currently only get evaluated at query compile time if all their operands are constants, with values known at query compile time. This is not the case here, so the condition will not be optimized away.
However, we could rewrite such conditions in the future as you are suggesting. Good idea!